### PR TITLE
[GJ-33] Upgrade substrait and generate substrait plan on driver

### PR DESCRIPTION
**Pull request by zzcclp**
1. Upgrade substrait to version [e1b4c04a1b](https://github.com/substrait-io/substrait/commit/e1b4c04a1b518912f4c4065b16a1b2c0ac8e14cf)  
2. Generate substrait plan on driver
3. Add TPCH test data and queries

Following PR:
1. NativeWholeStageRowRDD
2. JNI interface
3. Support both row and columnar iterator

Closes https://github.com/oap-project/gazelle-jni/issues/33.

### CODE REVIEW

File: jvm/src/main/java/com/intel/oap/substrait/expression/SelectionNode.java [hunk 1]
```diff
@@ -17,27 +17,29 @@
...
+import io.substrait.proto.*;
 
 import java.io.Serializable;
 
 public class SelectionNode implements ExpressionNode, Serializable {
     private final Integer fieldIdx;
 
     SelectionNode(Integer fieldIdx) {
-        this.fieldIdx = fieldIdx;
+        this.fieldIdx = fieldIdx + 1;
```
Comment (rui-mo) [inline jvm/src/main/java/com/intel/oap/substrait/expression/SelectionNode.java:28]: Why adding one here?

Comment (zzcclp) [reply jvm/src/main/java/com/intel/oap/substrait/expression/SelectionNode.java:28]: In protobuf, the value 0 means this class was reset:
image
So set the start index to 1.

Function mappings are also set the start index to 1.

Comment (zzcclp) [reply jvm/src/main/java/com/intel/oap/substrait/expression/SelectionNode.java]: Fixed: the index starts from 0.

**Comment by rui-mo** [comment]
LGTM!

**Pull request merged**